File: Java/src/com/algorithms/week4/LinearSearchIterative.java
package com.algorithms.week4;

import java.util.Scanner;

public class LinearSearchIterative {

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int n = scan.nextInt();
		int[] arr = new int[n];
		for (int i = 0; i < n; i++)
			arr[i] = scan.nextInt();
		int key = scan.nextInt();

		int result = linearSearch(arr, key);
		System.out.println(result);
		scan.close();
	}

	public static int linearSearch(int[] arr, int key) {
		for (int i = 0; i < arr.length; i++) {
			if (arr[i] == key)
				return i;
		}
		return -1;
	}

}

File: C++/week4/LinearSearchIterative.cpp
#include<bits/stdc++.h>

using namespace std;

int linearSearch(int* arr, int length, int key) {
    for (int i = 0; i < length; i++) {
        if (arr[i] == key)
            return i;
    }
    return -1;
}

int main() {
    int n, key;
    cin>>n;
    int* arr = new int[n];
    for (int i = 0; i < n; i++)
        cin>>arr[i];
    cin>>key;

    int result = linearSearch(arr, n, key);
    cout<<result<<endl;
}
